File: pickr/lib/classes/player.dart
import 'package:pickr/classes/card.dart';

class Player {
  //
  List<GamingCard> _cards;
  String _id;
  int _gameScore;
  int _matchScore;

  Player({String id}) {
    _id = id;
    _cards = <GamingCard>[];
    _gameScore = 0;
    _matchScore = 0;
  }

  /// Getter method for [_cards].
  List<GamingCard> get hand => _cards;

  String get id => _id;

  /// Setter method for [_cards].
  set hand(List<GamingCard> cards) => _cards = cards;

  /// Drops the first card
  ///
  /// The user should decide which card to drop, this is for semplicity.
  GamingCard drop() => _cards.first;

  /// Getter method for [_gameScore].
  int get gameScore => _gameScore;

  /// Setter method for [_gameScore].
  set setGameScore(int gameScore) => _gameScore += gameScore;

  /// Getter method for [_matchScore].
  int get matchScore => _matchScore;

  /// Setter method for [_matchScore].
  set setMatchScore(int matchScore) => _matchScore += matchScore;

  /// Removes the first card of the hand
  GamingCard delete() => _cards.removeAt(0);

  /// Picks a card and adds it to the hand
  void pick(GamingCard card) => _cards.add(card);
}

File: pickr-java/src/pickr/classes/Player.java
package pickr.classes;


import java.util.ArrayList;

public class Player {
	
	private ArrayList<GamingCard> cards;
	
	private String id;
	
	private int gameScore;
	
	private int matchScore;
	
	public Player(String id) {
		this.id = id;
		this.cards = new ArrayList<GamingCard>();
		this.gameScore = 0;
		this.matchScore = 0;
	}

	
	public ArrayList<GamingCard> getCards() { return cards;	}

	public void hand(ArrayList<GamingCard> cards) { this.cards = cards;	}
	
	public GamingCard drop() { return this.cards.get(0); }

	public int getGameScore() {	return gameScore; }

	public void setGameScore(int gameScore) { this.gameScore += gameScore;	}

	public int getMatchScore() { return matchScore; }

	public void setMatchScore(int matchScore) {	this.matchScore = matchScore; }

	public String getId() {	return id;	}

	public void delete() { this.cards.remove(0); }
	
	public void pick(GamingCard card) { this.cards.add(card);  }

}

